composer/callbacks/checkpoint_saver.py

```diff
@@ -289,25 +289,26 @@ class CheckpointSaver(Callback):  # noqa: D101
 
     def __init__(
         self,
-        folder: str = '{run_name}/checkpoints',
-        filename: str = 'ep{epoch}-ba{batch}-rank{rank}.pt',
+        checkpoint_save_path: str = '{run_name}/checkpoints',
```

Comment:
These are already inside of `CheckpointSaver`, so prefer to keep the names as before. Seems redundant to preface everything with `checkpoint` in the context of this object.

Reply:
I like the prefixing and everyone seemed to think it was fine who looked at the design doc

Reply:
Personally, I'd rather have consistent names here with trainer. This isn't purely internal since users can make custom checkpoint savers, so it's useful to have the same names imo. I believe this was the intention in the original design doc

Reply:
Fundamentally the issue here is we need to avoid verbose argument names as much as possible, e.g. see the names in [PTL](https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.callbacks.ModelCheckpoint.html), HF (called it output_dir), and [Keras](https://keras.io/api/callbacks/model_checkpoint/). So our naming is an anti-pattern to the rest of the community.

Reply:
Yeah that's true. Thanks for raising this! Everyone else has more brevity and ours is an anti-pattern. I rather have extreme clarity even if the arguments are a bit redundant and clunky and insult the intelligence of some of the users, but I see why the other libraries went for brevity. As I mentioned below, let's consider removing the prefixes at the end of this PR series, but I'll keep as is for now.